### DISCUSSION TITLE
Trying to add ListingPopety "Width" that has no "possible_values" but "value_ids" are required

### DISCUSSION THREAD

**Discussion by Demiurg-ls** [etsy/open-api#1288] [Q&A] [answered]
Hi, 

I have stuck on adding Listing with "Width", "Length", "Depth" properites (defined for my Taxonomy Node: "Fine Art Ceramics" [6096]).
The property returned by getPropertiesByTaxonomyId reports: "possible_values": []

Putting "updateListingProperty" with { ... value_ids: [], values: ["15.0"] ...} ends with error "value_ids are required"
I tried to guess proper values as there is nothing about it in documentaion (at lesast i didn't find any tips):
- value_ids: [0] reports "value_id must be >= 1"
- value_ids: [1]  adds "Width" property with value "Black" :)

Did i miss something?

Any help would be strongly appreciated :)
Best regards,
L

**Comment by turboturtleD** [comment]
I would try adding the properties you want using the Etsy front end, and then using the API examine the listing for clues to make a proper request.

**Comment by Demiurg-ls** [comment]
Hi, 

Thanks for suggestion. 

I have tried to use same "value_ids" (one of spoted "width" one)  when updating all properties for my test draft listing.
I guess "continous" properties (those without possible_valuses) got some kind of id while creating. 
It is not the same value_id for all "Width" properites.
Putting property with specific value_id seems to set the intial value that was set while creating value_id (first put) despite the property modified (Width, Length, Color ...)

By they way in Etsy Panel I couldn't correct "Black" i have set to my listing Width with API - i had to remove the property using API.

At the moment I wonder if problem is not connected with ListingInventory - will check it tomorrow.

Regards,
L

**Comment by Demiurg-ls** [comment, marked as answer]
One of my tries seems to work.

At the moment adding Listing proeperty that has no predefined values  i use **property_id** as **value_id** (i guess any not used by predefined value_ids would work) .
The properties are added/updated properly. In case of "Width", "Depth", "Height" properties the specified **value_id** is preserved but in case of "Weight" new one is created (different for each property instance :) 

A few observations: 
* value_id is probably connected with value - when restoring 116 g - value_id is also restored. 
* value_id is "global" - all(?) my "116.00" Grams listings have the same value_id. 
* don't rely on scale_name (i guess others names as well). Taxonomy reports "Grams" but ListingProperty has " g". In case of Width, Height, Depth "Centimeters" dissapear at all [null]. In both cases scale_id is proper.
* listing.item_width (*length, *height, *width) seems to be ignored. Etsy panel uses data from properties.

I hope it will be useful for someone,
L
